intake/interface/source/defined_plots.py

```diff
@@ -128,15 +152,99 @@ def selected(self, selected):
         """When plot is selected set, make sure widget stays upto date"""
         self.select.value = selected
 
-    def callback(self, *events):
+    def plot_selected(self, *events):
         for event in events:
             if event.name == 'value':
                 self.pane.object = self._plot_object(event.new)
-            if event.name == 'options':
-                self.instructions.object = self.instructions_contents
+            self.custom.name = "Create" if str(self.select.value) == 'None' else "Edit"
+            self.edit_options.visible = self.custom.name == "Edit"
+
+    def name_changed(self, *events):
+        for event in events:
+            if event.name == "value_input":
+                # Empty name not allowed and name cannot already be in use
+                if len(event.new) == 0 or event.new in self.options:
+                    self.interact_save.disabled = True
+                else:
+                    self.interact_save.disabled = False
+
+    def cancel(self, _):
+        self.pane.object = self._plot_object(self.selected)
+        self.out[0] = self.pane
+        self.children[0].visible = True
```

Comment:
Is it worth keeping these children as named attributes, so its more clear which visibilities are set?

Reply:
Yes, actually, that will make things clearer. I will remove the copious comments and make these named attributes so the code is self-describing.